src/index.js

```diff
@@ -19,78 +21,75 @@ const defaultOptions = {
+        const resultHandler = (err, res) => {
+          if (err) { return reject(err); }
+          if (res !== ARGON2_OK) {
+            return reject(new Error(argon2ErrorMessage(res)));
+          }
+          return resolve(hashOutput);
+        };
+        hashRaw.async(timeCost, memoryCost, parallelism,
+                      password, password.length,
+                      salt, salt.length,
+                      hashOutput, hashLength,
+                      resultHandler);
+      });
+      return promise;
     },
 
     hash(...args) {
-      const [password, salt, options, cb] = parseArgs(args);
+      const [password, salt, options] = parseArgs(args);
       const { timeCost, memoryCost, parallelism, hashLength } = options;
       const encodedSize = argon2Encodedlen(timeCost, memoryCost, parallelism,
                                                    salt.length, hashLength);
       const outputBuffer = new Buffer(encodedSize);
-      const resultHandler = (err, res) => {
-        if (err) { return cb(err, null); }
-        if (res !== ARGON2_OK) {
-          return cb(new Error(argon2ErrorMessage(res)), null);
-        }
-        return cb(null, ref.readCString(outputBuffer, 0));
-      };
-      hashEncoded.async(timeCost, memoryCost, parallelism,
-                        password, password.length,
-                        salt, salt.length,
-                        hashLength,
-                        outputBuffer, outputBuffer.length,
-                        resultHandler);
+      const promise = new Promise((resolve, reject) => {
+        const resultHandler = (err, res) => {
+          if (err) { return reject(err); }
+          if (res !== ARGON2_OK) {
+            return reject(new Error(argon2ErrorMessage(res)));
+          }
+          return resolve(ref.readCString(outputBuffer, 0));
+        };
+        hashEncoded.async(timeCost, memoryCost, parallelism,
+                          password, password.length,
+                          salt, salt.length,
+                          hashLength,
+                          outputBuffer, outputBuffer.length,
+                          resultHandler);
+      });
+      return promise;
     },
 
-    verify(encoded, password, cb) {
+    verify(encoded, password) {
       const encodedBuffer = ref.allocCString(encoded || '');
       let parsedPassword = password;
       if (typeof password === 'string') { parsedPassword = new Buffer(password); }
-      verify.async(encodedBuffer, parsedPassword, parsedPassword.length, (err, res) => {
-        if (err) { return cb(err, null); }
-        if (res !== ARGON2_OK) {
-          return cb(new Error(argon2ErrorMessage(res)), null);
-        }
-        return cb(null, res);
+      const promise = new Promise((resolve, reject) => {
+        const resultHandler = (err, res) => {
+          if (err) { return reject(err); }
+          if (res !== ARGON2_OK) {
+            return reject(new Error(argon2ErrorMessage(res)));
```

Comment:
Maybe it isn't. It can be useful to just use `argon2i.verify(...).then(assume hash matches)` but I think it's also very useful to use a custom assert function/error (such as Koa's assert that uses http-error). I'll change it to simply resolve with a boolean.